### PR TITLE
Re-organize Configuration documentation

### PR DESCRIPTION
**Pull request by eaba**
Part fix for #5483 

## Changes

* Moved ~/articles/hocon/index.md --> ~/articles/configuration/hocon.md

* Moved ~/articles/concepts/configuration.md --> ~/articles/configuration/index.md

* Moved ~/articles/configuration/* (all of the module-specific configurations) --> ~/articles/configuration/modules/*

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET Documentation Guidelines](https://getakka.net/community/contributing/documentation-guidelines.html#organization-of-documentation.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [x] Design discussion issue #5483

### CODE REVIEW

File: docs/articles/concepts/configuration.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+   <head>
+      <title>Akka.NET Configuration</title>
+      <meta http-equiv = "refresh" content="1;url=../configuration/index.html" />
+   </head>
+   <body>
+      <p>This page has been moved to <a href="../configuration/index.html">Akka.NET Configuration</a>.</p>
```
Comment (Aaronontheweb) [inline docs/articles/concepts/configuration.html:8]: LGTM

File: docs/articles/configuration/toc.yml [hunk 2]
```diff
@@ -1,12 +1,18 @@
+- name: Index
```
Comment (Aaronontheweb) [inline docs/articles/configuration/toc.yml:1]: LGTM

File: docs/articles/hocon/Index.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
```
Comment (Aaronontheweb) [inline docs/articles/hocon/Index.html:1]: Capitalization - this file will probably need to be lowercase `index.html` since we could be hosting on Linux.

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Overall, it looks good to me but need to take care of that file capitalization issue and the toc.yml changes, similar to what we described here https://github.com/akkadotnet/akka.net/issues/5502

File: docs/articles/hocon/Index.html (hunk 3 above)
Comment (eaba) [reply docs/articles/hocon/Index.html:1]: Got it!

**Comment by eaba** [comment]
> @eaba looks like the file name still isn't capitalized correctly - this is actually a big navigation problem on Linux systems.
> 
> This will help you: https://stackoverflow.com/questions/10523849/changing-capitalization-of-filenames-in-git

Please you confirm with the new commit?

**Comment by Aaronontheweb** [comment]
It's still a problem. You might need to use `wsl` to make this change.

**Review by Aaronontheweb — APPROVED** [review]

**Pull request merged**